tests/run-make/libtest-thread-limit/rmake.rs

```diff
@@ -0,0 +1,42 @@
+// libtest used to panic if it hit the thread limit. This often resulted in spurious test failures
+// (thread 'main' panicked at 'called Result::unwrap() on an Err value: Os
+// { code: 11, kind: WouldBlock, message: "Resource temporarily unavailable" }' ...
+// error: test failed, to rerun pass '--lib').
+// Since the fix in #81546, the test should continue to run synchronously
+// if it runs out of threads. Therefore, this test's final execution step
+// should succeed without an error.
+// See https://github.com/rust-lang/rust/pull/81546
```

Comment:
Suggestion: we need to explain *why* we need to drop down to linux syscalls. AFAICT, we need to do so because something like:

- We need to emulate an environment for libtest where threads are exhausted and spawning new threads are guaranteed to fail. This was previously achieved by `ulimit` shell builtin that called out to `prlimit64` underneath to set resource limits (specifically thread number limits). Now that we don't have a shell, we need to implement that ourselves.

Based on cross-checking with [dash](https://git.kernel.org/pub/scm/utils/dash/dash.git/)'s implementation of `ulimit -p` (BSD 3-clause), dash also uses `setrlimit` syscall to implement `ulimit -p` (`RLIMIT_NPROC`).

We should also link to `setrlimit(2)`: <https://linux.die.net/man/2/setrlimit>.